test/model/test_evaluation.py

```diff
@@ -0,0 +1,222 @@
+    MASE(),
+    MSIS(),
+    ND(),
+    MeanWeightedSumQuantileLoss(quantile_levels=[0.1, 0.5, 0.9]),
+]
+
+
+def infer_metrics_df_shape(
+    metrics,
+    test_data: TestData,
+    axis: Union[int, Tuple[int]],
+) -> Tuple[int]:
+    """
+    Infer expected shape of metrics data frame.
+    """
+    labels = list(test_data.label)
+    test_data_shape = (len(labels), *labels[0]["target"].shape)
+
+    if axis is None:
+        axis = tuple(range(len(test_data_shape)))
+    if isinstance(axis, int):
+        axis = (axis,)
+
+    return (
+        prod(
+            test_data_shape[d]
+            for d in range(len(test_data_shape))
+            if d not in axis
+        ),
+        len(metrics),
+    )
+
+
+@pytest.mark.parametrize(
+    "metrics, test_data, axis",
+    [
+        (
+            _test_metrics,
+            split(_test_univariate_dataset, offset=-12)[1].generate_instances(
+                prediction_length=3, windows=4
+            ),
+            None,
+        ),
+        (
+            _test_metrics,
+            split(_test_univariate_dataset, offset=-12)[1].generate_instances(
+                prediction_length=3, windows=4
+            ),
+            (0, 1),
+        ),
+        (
+            _test_metrics,
+            split(_test_univariate_dataset, offset=-12)[1].generate_instances(
+                prediction_length=3, windows=4
+            ),
+            0,
+        ),
+        (
+            _test_metrics,
+            split(_test_univariate_dataset, offset=-12)[1].generate_instances(
+                prediction_length=3, windows=4
+            ),
+            1,
+        ),
+        (
+            _test_metrics,
+            split(_test_multivariate_dataset, offset=-12)[
+                1
+            ].generate_instances(prediction_length=3, windows=4),
+            None,
+        ),
+        (
+            _test_metrics,
+            split(_test_multivariate_dataset, offset=-12)[
+                1
+            ].generate_instances(prediction_length=3, windows=4),
+            (0, 1, 2),
+        ),
+        (
+            _test_metrics,
+            split(_test_multivariate_dataset, offset=-12)[
+                1
+            ].generate_instances(prediction_length=3, windows=4),
+            (0, 2),
+        ),
+        (
+            _test_metrics,
+            split(_test_multivariate_dataset, offset=-12)[
+                1
+            ].generate_instances(prediction_length=3, windows=4),
+            (0, 1),
+        ),
+        (
+            _test_metrics,
+            split(_test_multivariate_dataset, offset=-12)[
+                1
+            ].generate_instances(prediction_length=3, windows=4),
+            (1, 2),
+        ),
+    ],
+)
+def test_evaluation_shape(
+    metrics,
+    test_data: TestData,
+    axis: Union[int, Tuple[int]],
+):
+    num_samples = 31
+
+    sample_forecasts = [
+        SampleForecast(
+            samples=np.zeros((num_samples, *label["target"].shape)),
```

Comment:
I think the samples shouldn't be constructed based on label shape (C, T). SampleForecast for multivariate follow (N, T, C).

Reply:
And this is indeed pretty annoying